### PR TITLE
preload function is now under the render property

### PR DESCRIPTION
**Pull request by ricardo-cantu**
latest Next release changed object shape

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ function createFactory(moduleName) {
         options = mockOptions;
       }
       const LoadableComponent = dynamic(loader, options);
-      mockInitializers.push(() => LoadableComponent.preload());
+      mockInitializers.push(() => LoadableComponent.render.preload());
```
Comment (exogen) [inline index.js]: We don't want to be tied to an exact Next.js version necessarily – how about checking for both `LoadableComponent.preload` and `LoadableComponent.preload .render.preload`?

**Comment by kylecesmat** [comment]
@exogen I've tested this fix against a next@9 application and it works great 🎉

File: index.js [hunk 2]
```diff
@@ -31,7 +31,10 @@ function createFactory(moduleName) {
         options = mockOptions;
       }
       const LoadableComponent = dynamic(loader, options);
-      mockInitializers.push(() => LoadableComponent.preload());
+      mockInitializers.push(() =>
+        LoadableComponent.preload
+          ? LoadableComponent.preload()
+          : LoadableComponent.render.preload()
```
Comment (kylecesmat) [inline index.js]: Noticed a type error: missing closing `)` for the `push` method

**Comment by exogen** [comment]
Hmm, I can't get this to work when I upgrade the devDeps to use next@9.1.2.

```
$ jest
 FAIL  ./index.test.js
  ● Test suite failed to run

    Cannot find module 'next-server/dynamic' from 'NextDynamicNested.js'

    However, Jest was able to find:
    	'test/NextDynamicNested.js'

    You might want to include a file extension in your import, or update your 'moduleFileExtensions', which is currently ['js', 'json', 'jsx', 'ts', 'tsx', 'node'].

    See https://jestjs.io/docs/en/configuration#modulefileextensions-array-string

      1 | import React from "react";
    > 2 | import dynamic from "next/dynamic";
        | ^
      3 |
      4 | const NextDynamic = dynamic(
      5 |   () =>

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:203:17)
      at Object.<anonymous> (test/NextDynamicNested.js:2:1)
```

It's complaining about `next-server/dynamic` which may not exist anymore, but that's why we use `virtual: true` on the mock. Even when I take out the `next-server/dynamic` mock, it still throws that error. Any ideas? @kylecesmat @ricardo-cantu

**Comment by exogen** [comment]
Excitingly, upgrading Jest seems to fix that test suite issue...so I guess a Jest bug?

**Comment by exogen** [comment]
This is now published in v1.0.1.

**Pull request merged**